src/org/exist/xquery/InlineFunction.java

```diff
@@ -96,6 +96,12 @@ public int returnsType() {
     public void resetState(boolean postOptimization) {
         super.resetState(postOptimization);
         // clear closure variables set on inline function
-        function.setClosureVariables(null);
+        calls.forEach(call -> {
```

Comment:
I think this could be reformulated to be more efficient, i.e.:

```java
if (!postOptimization) {
  calls.forEach(call -> call.getFunction().setClosureVariables(null));
}
```